class-components/src/components/Floyt/Flyout.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { AppDispatch, RootState } from '../../store/store';
+import { unselectAll } from '../../store/searchSlice';
+import { People } from './Flyout.type';
+import './Flyout.css';
+
+const Flyout: React.FC = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const selectedItems = useSelector((state: RootState) => state.search.selectedItems);
+  const items = useSelector((state: RootState) => state.search.items);
+
+  const handleUnselectedAll = () => {
+    dispatch(unselectAll());
+  };
+
+  const handleDownload = () => {
+    const selectedPeople = items.filter((item: People) => selectedItems.includes(item.url));
+    const csvHeaders = ['Name', 'Details URL', 'Height', 'Eye color', 'Skin color'];
+    const csvRows = [
+      csvHeaders,
+      ...selectedPeople.map((item) => [
+        item.name,
+        item.url,
+        item.height,
+        item.eye_color,
+        item.skin_color,
+      ]),
+    ];
+    const csvString = csvRows.map((row) => row.join(', ')).join('\n');
+    const blob = new Blob([csvString], { type: 'text/csv;charset=utf-8' });
+    const link = document.createElement('a');
+    link.href = URL.createObjectURL(blob);
+    link.download = `${selectedItems.length}_person(s).csv`;
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
```

Comment:
Are you sure there's no way to do it other than manipulate DOM directly with document.createElement('a'), appendChild, removeChild? Direct DOM manipulation is heavily discouraged and should be reserved mainly for external libraries usage where it cannot be avoided. 

**Please re-do it so there is no direct DOM manipulation** - if you're not sure how, let me know and we'll try to think of something together